### ISSUE TITLE
P4C doesn't build anymore in container

### ISSUE THREAD

**Issue by stolsma** [ipdk-io/ipdk#140]
Building P4C fails with : 

```
#17 2043.5 error: patch failed: control-plane/p4RuntimeSerializer.cpp:148
#17 2043.5 error: control-plane/p4RuntimeSerializer.cpp: patch does not apply
```

This is probably caused by changing the used hash of p4c repo... Will investigate if the patch is still needed!  

See full log:
```
#17 2032.6 Removing P4C directory if it already exits
#17 2032.6 Cloning  P4C repo
#17 2032.6 Num cores on a system: 2
#17 2032.6 Number of Parallel threads used: -j2 ...
#17 2032.6 
#17 2032.6 Cloning into 'P4C'...
#17 2039.4 Submodule 'backends/ebpf/runtime/contrib/libbpf' (https://github.com/libbpf/libbpf) registered for path 'backends/ebpf/runtime/contrib/libbpf'
#17 2039.4 Submodule 'control-plane/p4runtime' (https://github.com/p4lang/p4runtime.git) registered for path 'control-plane/p4runtime'
#17 2039.4 Submodule 'test/frameworks/gtest' (https://github.com/google/googletest.git) registered for path 'test/frameworks/gtest'
#17 2039.4 Cloning into '/root/P4C/backends/ebpf/runtime/contrib/libbpf'...
#17 2040.3 Cloning into '/root/P4C/control-plane/p4runtime'...
#17 2041.2 Cloning into '/root/P4C/test/frameworks/gtest'...
#17 2043.0 Submodule path 'backends/ebpf/runtime/contrib/libbpf': checked out '3b239425426e4fa1c204ea3c708d36ec3f509702'
#17 2043.0 Submodule path 'control-plane/p4runtime': checked out '7a322f35f0c80bf20bc7fcc96f9d1ab77e5fd07a'
#17 2043.2 Submodule path 'test/frameworks/gtest': checked out 'aa148eb2b7f70ede0eb10de34b6254826bfb34f4'
#17 2043.5 Note: switching to '4837beadf3fc390dd3a39434fec4fa6658e993be'.
#17 2043.5 
#17 2043.5 You are in 'detached HEAD' state. You can look around, make experimental
#17 2043.5 changes and commit them, and you can discard any commits you make in this
#17 2043.5 state without impacting any branches by switching back to a branch.
#17 2043.5 
#17 2043.5 If you want to create a new branch to retain commits you create, you may
#17 2043.5 do so (now or later) by using -c with the switch command. Example:
#17 2043.5 
#17 2043.5   git switch -c <new-branch-name>
#17 2043.5 
#17 2043.5 Or undo this operation with:
#17 2043.5 
#17 2043.5   git switch -
#17 2043.5 
#17 2043.5 Turn off this advice by setting config variable advice.detachedHead to false
#17 2043.5 
#17 2043.5 HEAD is now at 4837beadf Handle struct expression in ebfp backend (#3173)
#17 2043.5 error: patch failed: control-plane/p4RuntimeSerializer.cpp:148
#17 2043.5 error: control-plane/p4RuntimeSerializer.cpp: patch does not apply
#17 2043.5 /root/p4-sde/install
#17 2043.5
```

**Comment by mestery** [comment]
I merged #139, let me know if this still fails with that PR in mainline now or not and we can figure this out.

**Comment by stolsma** [comment]
This works now after #139....
